qemu/tests/cfg/block_resize.cfg

```diff
@@ -4,61 +4,69 @@
-        disk_update_cmd += " && echo exit >> cmd"
-        disk_update_cmd += " && diskpart /s cmd"
-        # "::" is used to separate the commands as it is not used in both Linux and Windows
-        disk_update_cmd += "::"
-        disk_update_cmd += "echo select disk 1 > cmd"
-        disk_update_cmd += " && echo select partition 1 >> cmd"
-        disk_update_cmd += " && echo shrink desired=DISK_CHANGE_SIZE >> cmd"
-        disk_update_cmd += " && echo exit >> cmd && diskpart /s cmd"
-        disk_unit = M
-        disk_rescan_cmd = "echo rescan > cmd"
-        disk_rescan_cmd += " && echo exit >> cmd"
-        disk_rescan_cmd += " && diskpart /s cmd"
+    accept_ratio = 0.005
+    iozone_path = "WIN_UTILS:\Iozone\iozone.exe"
+    iozone_option = " -azR -r 64k -n 512M -g 1G -M -I -b iozone.xls -f %s:\testfile"
+    Linux:
+        labeltype = msdos
+        block_size_cmd = "fdisk -l | grep {0}"
+        block_size_pattern = ",\s+(\d+\s+b)ytes"
+        iozone_option = " -azR -r 64k -n 512M -g 1G -I -f %s/testfile"
+    qcow2:
+        Host_RHEL.m6, Host_RHEL.m7.u0, Host_RHEL.m7.u1, Host_RHEL.m7.u2, Host_RHEL.m7.u3, Host_RHEL.m7.u4, Host_RHEL.m7.u5:
+            disk_change_ratio = "1.5"
+    variants:
+        - with_data_disk:
+            images += " stg"
+            image_size_stg = 10G
+            image_name_stg = images/stg
+            create_image_stg = yes
+            remove_image_stg = yes
+            force_create_image_stg = yes
+            image_snapshot_image = yes
+            blk_extra_params_stg = "serial=TARGET_DISK0"
+            image_extra_params = ""
+            format_disk = yes
+            Host_RHEL.m6..ide:
+                blk_extra_params_stg = "wwn=0x5000123456789abc"
+            Windows:
+                block_size_pattern = "1\s+(\d+)"
+                need_rescan = yes
+            variants:
+                - fmt_qcow2:
+                    image_format_type = qcow2
+                    image_format_stg = qcow2
+                - fmt_raw:
+                    image_format_type = raw
+                    image_format_stg = raw
+        - with_system_disk:
+            only Windows
+            blk_extra_params_image1 = "serial=TARGET_DISK0"
+            Host_RHEL.m6..ide:
+                blk_extra_params_image1 = "wwn=0x5000123456789abc"
+            backup_image_before_testing = yes
+            restore_image_after_testing = yes
+            Windows:
+                block_size_pattern = "0\s+(\d+)"
+                need_rescan = yes
     ide:
-        # TODO this is bug that ide not support qcow2 format block resize
-        # recently. Will remove this line when the bug fixed.
-        no qcow2
+        need_reboot = yes
+    ahci:
         need_reboot = yes
     virtio_blk:
-        need_reboot = no
         driver_name = viostor
     virtio_scsi:
-        need_reboot = no
         driver_name = vioscsi
         Linux:
             guest_prepare_cmd = "echo 1 > /sys/block/sda/device/rescan"
```

Comment:
if resize data disk, I thinks the re-scanned disk is not \"sda\"

Reply:
updated, thanks~